### DISCUSSION TITLE
Crazyradio PA Connection Problem

### DISCUSSION THREAD

**Discussion by b1naryTR** [bitcraze/discussions#977] [Q&A]
On both Windows and Linux my Crazyradio PA doesn't show up. There is no new device on lsusb on linux. I think I corrupted the firmware somehow, is there any way to restore it? Official documentation states that I need an Bus Pirate [here](https://www.bitcraze.io/documentation/repository/crazyradio-firmware/master/development/spi_programming/). Can I get away with an cheap USB-JTAG or USB-TTL adapter? Like [this ](https://www.direnc.net/usb-cpld-fpga-jtag-programlayici-en)or [this](https://www.direnc.net/cp2102-6-pin-usb-20-uart-ttl-seri-donusturucu-arduino-modul-en)?

**Comment by knmcguire** [comment]
hi! Just to be clear, you are talking about the [crazyradio PA ](https://store.bitcraze.io/collections/accessories/products/crazyradio-pa), not the [crazyradio 2.0](https://store.bitcraze.io/collections/accessories/products/crazyradio-2-0). 

In case of the crazyradio PA version 1: Before starting buspirate or anything, as that is a quite drastic measure since it removes everything, you should check if the bootloader still works and if you can [reflash the firmware with binutils](https://www.bitcraze.io/documentation/repository/crazyradio-firmware/master/building/building_flashing/).

**Reply by b1naryTR** [reply]
Only red led lits up on plug-in, and I couldn't get the thing even recognised on any device I tried, it just doesn't show up. I am thinking it's an bootloader problem.

**Reply by knmcguire** [reply]
alright, so it is crazyradio PA :). @ataffanel can help you further in the other thread.

**Comment by ataffanel** [comment]
The best way to know if a Crazyradio PA needs to be re-flashed is to look at the LED on startup: if both RED and GREEN LEDs lit up and then go off when connecting to a PC, then the firmware is working and your connection problem might be software related.

If not, then you need to re-flash using the SPI port. Anything that talks SPI could work, I have experimented with using a raspberry-pi for example.

The easiest would be to try to fix the bootloader with this script: https://github.com/bitcraze/crazyradio-firmware/tree/master/fix_bootloader. It works on a raspberrypi which you might have arround. The code is not very complex and could be ported to a 3.3V Arduino for exampe or anything else that can implement SPI or GPIO at 3.3V.

**Reply by b1naryTR** [reply]
Hey there, tried the script with my Pi Zero W and at first my PA launched into bootloader mode, then I flashed the latest official firmware (v0.53) after that everything went well until I flashed a broken firmware again, as I was trying to get the PA in bootloader mode again with my Pi, same thing started to happen at start, when I use the script red light stays open, and I can't use bootloader mode for some reason. 

So, everything was working fine at first, but then I can't even load into bootloader mode with my Pİ.

**Reply by ataffanel** [reply]
The fix bootloader script only writes the instruction "JMP booloader" at the start of the flash. This is what makes it so fast. The problem is that if you erase the bootloader (which is possible since it is not protected, writing our firmware should never do that but writing third party firmware might), then this procedure is not working anymore.

It is likely you will now have to re-flash the bootloader using SPI. Since you seem to be working with firmware quite actively, I suggest you setup an SPI programmer, either by using the FTDI scripts with a FTDI-based jtag probe or with your Raspberry-pi: the recovery script can be addapted to flash the full flash. The nRF24lu1 behaves like an SPI flash so it is not hard to make a script that can write to it. With an SPI programmer you will not have to care anymore about bricking the bootloader.

To re-flash the bootloader I know there is open-source version of nRF24 bootloaders, I quickly found at least 2 of those, a DFU bootloader: https://github.com/ranma/nrf24lu1-dfu-bootloader and one compatible with the nordic protocol: https://github.com/ahtn/nrf24lu1p-512-bootloader.

**Reply by b1naryTR** [reply]
I have never worked with Crazyradio nor SPI over Raspberry Pi. I was planning to make an presentation about Mousejacking on my school project. I really need more specific help about writing to this thing.

Weird thing is, I don't think Mousejack firmware writes to bootloader. It happened after I updated the firmware with official firmware. Does the official firmware include some kind of new bootloader?

**Reply by tobbeanton** [reply]
There are arduino based programmers like the [NRF24LU1prog](https://github.com/FMMT666/NRF24LU1prog#nrf24lu1prog) you can check out.
